### PR TITLE
Create SDK for version 2024-06-preview

### PR DESCRIPTION
**Pull request by emmeliaAra**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by jsquire — DISMISSED** [review]
@emmeliaAra :

Your account lacks the required public GitHub organizations required of an internal contributor.  Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:910]: @ronniegeraghty:  Looks like there's a proposed new tag.   I'm going to suggest that we add spaces for readability:  "Hardware Security Modules".

Please offer thoughts.  If approved, I'll push it out to all repositories.

**Comment by emmeliaAra** [comment]
> @emmeliaAra :
> 
> Your account lacks the required public GitHub organizations required of an internal contributor. Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
> 
> * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> 
> You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.
> 
> Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

@jsquire Updated visibility for orgs and added my username in Codeowners file.

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:910]: @emmeliaAra:  Please move this down into the Management section below, preserving alphabetical ordering by tag.

You're also missing the PRLabel block.   Please follow the format of existing entries below.

For example:
```suggestion
# PRLabel: %Hardware Security Modules
/sdk/hardware/secirutymodules/Azure.ResourceManager.*/             @emmeliaAra 
# ServiceLabel: %Hardware Security Modules %Mgmt
# ServiceOwners:                                                   @emmeliaAra
```

Comment (emmeliaAra) [reply .github/CODEOWNERS:910]: Updated service name and moved it to the 'H' section

Comment (emmeliaAra) [reply .github/CODEOWNERS]: updated service Label to "Hardware Security Modules"

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:534]:
```suggestion
# ServiceLabel: %Hardware Security Modules %Mgmt
```

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:532]:
```suggestion
sdk/hardwaresecuritymodules/Azure.ResourceManager.*/               @emmeliaAra
```

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:531]: We'll want to move this entire block down to the management section.   Since we're organizing alphabetically by label, it should land around L956.

**Comment by jsquire** [comment]
> @jsquire Updated visibility for orgs and added my username in Codeowners file.

Confirmed.   Account updates look good.

File: .github/CODEOWNERS
Comment (emmeliaAra) [reply .github/CODEOWNERS]: updated

File: .github/CODEOWNERS
Comment (emmeliaAra) [reply .github/CODEOWNERS]: updated!

**Comment by emmeliaAra** [comment]
@jsquire  the 2 checks below fail with spelling error about "statu". but in the swagger we have status. Do I have to manually update it ?

[Swagger link](https://github.com/emmeliaAra/azure-rest-api-specs/blob/26c8a881f6b1e8a1c932344ba8f80d1dcbc4a10b/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json#L1001) 

- Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci
- Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci (Build Analyze)


2/2 /mnt/vss/_work/1/s/sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs 386.37ms X
CSpell: Files checked: 2, Issues found: 4 in 1 files
cspell run complete, restoring original configuration and removing temp file.
./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:47:130 - Unknown word (Statu)
./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:48:159 - Unknown word (Statu)
./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:51:132 - Unknown word (Statu)
./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:52:161 - Unknown word (Statu)
##[error]Spelling errors detected. To correct false positives or learn about spell checking see: https://aka.ms/azsdk/engsys/spellcheck

**Comment by jsquire** [comment]
> @jsquire the 2 checks below fail with spelling error about "statu". but in the swagger we have status. Do I have to manually update it ?
> 
> [Swagger link](https://github.com/emmeliaAra/azure-rest-api-specs/blob/26c8a881f6b1e8a1c932344ba8f80d1dcbc4a10b/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json#L1001)
> 
> * Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci
> * Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci (Build Analyze)
> 
> 2/2 /mnt/vss/_work/1/s/sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs 386.37ms X CSpell: Files checked: 2, Issues found: 4 in 1 files cspell run complete, restoring original configuration and removing temp file. ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:47:130 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:48:159 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:51:132 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:52:161 - Unknown word (Statu) ##[error]Spelling errors detected. To correct false positives or learn about spell checking see: https://aka.ms/azsdk/engsys/spellcheck

This one I'll need to defer to @ArthurMa1978 for.

**Comment by ArthurMa1978** [comment]
> @jsquire the 2 checks below fail with spelling error about "statu". but in the swagger we have status. Do I have to manually update it ?
> 
> [Swagger link](https://github.com/emmeliaAra/azure-rest-api-specs/blob/26c8a881f6b1e8a1c932344ba8f80d1dcbc4a10b/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json#L1001)
> 
> * Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci
> * Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci (Build Analyze)
> 
> 2/2 /mnt/vss/_work/1/s/sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs 386.37ms X CSpell: Files checked: 2, Issues found: 4 in 1 files cspell run complete, restoring original configuration and removing temp file. ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:47:130 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:48:159 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:51:132 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:52:161 - Unknown word (Statu) ##[error]Spelling errors detected. To correct false positives or learn about spell checking see: https://aka.ms/azsdk/engsys/spellcheck

Fixed, it caused by the name plural "optimization".

**Comment by emmeliaAra** [comment]
> > @jsquire the 2 checks below fail with spelling error about "statu". but in the swagger we have status. Do I have to manually update it ?
> > [Swagger link](https://github.com/emmeliaAra/azure-rest-api-specs/blob/26c8a881f6b1e8a1c932344ba8f80d1dcbc4a10b/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json#L1001)
> > 
> > * Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci
> > * Azure Pipelines / net - hardwaresecuritymodules - mgmt - ci (Build Analyze)
> > 
> > 2/2 /mnt/vss/_work/1/s/sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs 386.37ms X CSpell: Files checked: 2, Issues found: 4 in 1 files cspell run complete, restoring original configuration and removing temp file. ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:47:130 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:48:159 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:51:132 - Unknown word (Statu) ./sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/api/Azure.ResourceManager.HardwareSecurityModules.netstandard2.0.cs:52:161 - Unknown word (Statu) ##[error]Spelling errors detected. To correct false positives or learn about spell checking see: https://aka.ms/azsdk/engsys/spellcheck
> 
> Fixed, it caused by the name plural "optimization".

Thanks !

File: .github/CODEOWNERS
Comment (emmeliaAra) [reply .github/CODEOWNERS]: updated

**Review by ArthurMa1978 — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**
